sklearn/semi_supervised/_self_training.py

```diff
@@ -207,8 +207,11 @@ def fit(self, X, y):
 
             if self.n_iter_ == 1:
                 # Only validate in the first iteration so that n_iter=0 is
-                # equivalent to the base_estimator itself.
-                _validate_estimator(self.base_estimator)
+                # equivalent to the base_estimator_ itself.
```

Comment:
@oliverrausch Do you recall what is the meaning of this line and the previous one?
It was not obvious to me.

Reply:
May I suggest:

```
# Only validate the fitted estimator of the first iteration iteration.
```

Actually since the validation is really cheap, I would not mind simplifying this code and removing the `if self.n_iter_ == 1:` condition.